kupyna/src/lib.rs

```diff
@@ -0,0 +1,331 @@
+// #![no_std]
+#![doc = include_str!("../README.md")]
+#![doc(
+    html_logo_url = "https://raw.githubusercontent.com/RustCrypto/media/6ee8e381/logo.svg",
+    html_favicon_url = "https://raw.githubusercontent.com/RustCrypto/media/6ee8e381/logo.svg"
+)]
+#![cfg_attr(docsrs, feature(doc_auto_cfg))]
+#![forbid(unsafe_code)]
+#![warn(missing_docs)]
+
+pub use digest::{self, Digest};
+
+use core::{convert::TryInto, fmt};
+#[cfg(feature = "zeroize")]
+use digest::zeroize::{Zeroize, ZeroizeOnDrop};
+use digest::{
+    HashMarker, InvalidOutputSize, Output,
+    block_buffer::Eager,
+    core_api::{
+        AlgorithmName, Block, BlockSizeUser, Buffer, BufferKindUser, CoreWrapper,
+        CtVariableCoreWrapper, OutputSizeUser, RtVariableCoreWrapper, TruncSide, UpdateCore,
+        VariableOutputCore,
+    },
+    crypto_common::hazmat::{DeserializeStateError, SerializableState, SerializedState},
+    typenum::{U6, U28, U32, U48, U64, U72, U128, U136, Unsigned},
+};
+
+mod compress1024;
+mod compress512;
+mod tables;
+
+/// Lowest-level core hasher state of the short Kupyna variant.
+#[derive(Clone)]
+pub struct KupynaShortVarCore {
```

Comment:
It may be worth to move `KupynaShortVarCore`/`KupynaLongVarCore` into separate modules (e.g. `short.rs` and `long.rs`) and keep in `lib.rs` only re-exports and type alias definitions.